website/project/views/contributor.py

```diff
@@ -669,6 +670,12 @@ def claim_user_form(auth, **kwargs):
     user.update_guessed_names()
     # The email can be the original referrer email if no claimer email has been specified.
     claimer_email = unclaimed_record.get('claimer_email') or unclaimed_record.get('email')
+
+    # If there is a registered user with this email, redirect to 're-enter password' page
+    user_from_email = User.find_one(Q('username', 'eq', claimer_email))
```

Comment:
What about the edge case to this edge case where a user has confirmed this address but it's not their primary? i.e. `Q('emails', 'eq', claimer_email)`